db/db_iter.cc

```diff
@@ -295,8 +295,8 @@ bool DBIter::FindNextUserEntryInternal(bool skipping_saved_key,
               assert(ikey_.type != kTypeBlobIndex);
               if (ikey_.sequence >= start_seqnum_) {
                 saved_key_.SetInternalKey(ikey_);
-                valid_ = true;
-                return true;
+                valid_ = iter_.PrepareValue();
```

Comment:
What would be the hard if we call `iter_.PrepareValue()` before the switch? It's true that some code path it can be prevented. But these paths are low chance paths and even if we make them a little bit more expensive, it should not matter much. But in that way, it's less prone to bugs in the future, when we touch the code in switch. Thoughts?